x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrStatsAction.java

```diff
@@ -0,0 +1,156 @@
+                                builder.field("leader_max_seq_no", status.leaderMaxSeqNo());
+                                builder.field("follower_global_checkpoint", status.followerGlobalCheckpoint());
+                                builder.field("follower_max_seq_no", status.followerMaxSeqNo());
+                                builder.field("last_requested_seq_no", status.lastRequestedSeqNo());
+                                builder.field("number_of_concurrent_reads", status.numberOfConcurrentReads());
+                                builder.field("number_of_concurrent_writes", status.numberOfConcurrentWrites());
+                                builder.field("number_of_queued_writes", status.numberOfQueuedWrites());
+                                builder.humanReadableField(
+                                        "total_fetch_time_millis",
+                                        "total_fetch_time",
+                                        new TimeValue(status.totalFetchTimeNanos(), TimeUnit.NANOSECONDS));
+                                builder.field("number_of_successful_fetches", status.numberOfSuccessfulFetches());
+                                builder.field("number_of_fetches", status.numberOfSuccessfulFetches() + status.numberOfFailedFetches());
+                                builder.field("operations_received", status.operationsReceived());
+                                builder.humanReadableField(
+                                        "total_transferred_bytes",
+                                        "total_transferred",
+                                        new ByteSizeValue(status.totalTransferredBytes(), ByteSizeUnit.BYTES));
+                                builder.humanReadableField(
+                                        "total_index_time_millis",
+                                        "total_index_time",
+                                        new TimeValue(status.totalIndexTimeNanos(), TimeUnit.NANOSECONDS));
+                                builder.field("number_of_successful_bulk_operations", status.numberOfSuccessfulBulkOperations());
+                                builder.field(
+                                        "number_of_bulk_operations",
+                                        status.numberOfSuccessfulBulkOperations() + status.numberOfFailedBulkOperations());
+                                builder.field("number_of_operations_indexed", status.numberOfOperationsIndexed());
+                            }
+                            builder.endObject();
+                        }
+                    }
+                    builder.endObject();
+                }
+            }
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    public static class TasksRequest extends BaseTasksRequest<TasksRequest> {
+
+        private String indexName;
+
+        public void setIndexName(final String indexName) {
```

Comment:
I think we should follow the same patterns as an indices stats request - i.e., we should except a list of indices or a wild card pattern.

Reply:
I think this will require some effort. I will look into this tomorrow.

Reply:
@bleskes I pushed 2ee14e31d8230afad6b34f224c03ed97a0cf66cd. I am not a fan of it since the `TasksRequest#match` method is now a lie so that we can defer index resolution to the transport action. This is a limitation of the tasks API that puts us here. I do not think we should block CCR on addressing this so we have this workaround for now. I dislike it but maybe it's tolerable for now to give us an API consistent with others. Let me know what you think.